src/main/java/io/nats/client/impl/Headers.java

```diff
@@ -581,13 +585,31 @@ public boolean isReadOnly() {
 	@Override
 	public boolean equals(Object o) {
 		if (this == o) return true;
-		if (o == null || getClass() != o.getClass()) return false;
+		if (!(o instanceof Headers)) return false;
 		Headers headers = (Headers) o;
 		return Objects.equals(valuesMap, headers.valuesMap);
 	}
 
 	@Override
 	public int hashCode() {
-		return Objects.hash(valuesMap);
+		return Objects.hashCode(valuesMap);
+	}
+
+	@SuppressWarnings("deprecation")
+	@Override
+	public String toString() {
+		byte[] b = getSerialized();
+		if (b.length <= HVCRLF_BYTES + 2){
+			return "";
+		}
+
+		for (int i = 0, len = b.length; i < len; i++) {
+			if (b[i] == LF) {
+				b[i] = ' ';
+			} else if (b[i] == CR) {
+				b[i] = ';';
+			}
+		}
+		return new String(b, 0, HVCRLF_BYTES, b.length - HVCRLF_BYTES - 3);
```

Comment:
I'm fully on board with squeezing every nano second out, but we can't use a deprecated method. So just simplify this. We are talking maybe nano seconds on an operation that probably isn't used all that much. Headers have been around for years and no one has asked for this, users probably write their own anyway.

```
@Override
public String toString() {
	byte[] b = getSerialized();
	for (int i = 0, len = b.length; i < len; i++) {
		switch (b[i]) {
			case CR: b[i] = ';'; break;
			case LF: b[i] = ' '; break;
		}
	}
	return new String(b, US_ASCII);
}
```